### DISCUSSION TITLE
Include font-face

### DISCUSSION THREAD

**Discussion by ballotade** [dompdf/dompdf#3025] [Q&A]
Hello,

how can I include a custom font with css in dompdf.

I tried this:
`@font-face { font-family: Questrial; src: url("Questrial-Regular.ttf"); font-weight: normal; font-style: normal; }`

It worked with dompdf 0.8
But with the newest version (I think it is version 2.0) it does not work.

Can I include it with base64?

Thanks for your ideas. :)

**Comment by bsweeney** [comment]
base64 isn't currently supported, though I'm working on improving at-font-face for an upcoming release and it should be addressed at that time.

Do you receive any errors? The main change with 2.0.x was to address potential security vulnerabilities by restricting file loading in the at-font-face declaration. If this is a reference to a filesystem location you'll want to make sure that's allowed by the chroot Option. If this is a HTTP-based reference you'll want to make sure that the isRemoteAllowed Option is true.

You can try to get more info on why this fails by following the [debug guidance in the wiki](https://github.com/dompdf/dompdf/wiki/Debugging-Dompdf).

**Reply by steinkerchner** [reply]
I just ran into a problem like this and solved it by removing the dash from the filename. A space in the filename also caused a problem. It did not fail to generate a PDF, but the generated PDF was messed up because it did not have a valid font included. When I opened the generated PDF in Adobe Acrobat, it gave an error that the font "contains a bad /BBox."
